src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/MemberAndDeclarationInfo.cs

```diff
@@ -246,8 +247,71 @@ private sealed class MemberNameComparer : IComparer<MemberAndDeclarationInfo>
             public int Compare(MemberAndDeclarationInfo x, MemberAndDeclarationInfo y)
             {
                 var comp = string.Compare(x.Name, y.Name, StringComparison.Ordinal);
-                return comp != 0 ? comp : (y._inheritanceLevel - x._inheritanceLevel);
+                if (comp == 0)
+                {
+                    return y._inheritanceLevel - x._inheritanceLevel;
+                }
+
+                // Data members come first
+                var xHoldsData = HoldsData(x);
+                var yHoldsData = HoldsData(y);
+                if (xHoldsData && yHoldsData)
+                {
+                    return comp;
+                }
+                if (xHoldsData)
+                {
+                    return -1;
+                }
+                if (yHoldsData)
+                {
+                    return 1;
+                }
+
+                return comp;
+
+                static bool HoldsData(MemberAndDeclarationInfo member)
+                {
+                    if (member.MemberType == MemberTypes.Field)
+                    {
+                        return true;
+                    }
+
+                    if (member._member is PropertyInfoImpl propertyInfo)
+                    {
+                        return IsNullOrCompilerGenerated(propertyInfo.Property.GetMethod) && IsNullOrCompilerGenerated(propertyInfo.Property.SetMethod);
```

Comment:
>IsNullOrCompilerGenerated(propertyInfo.Property.GetMethod) && IsNullOrCompilerGenerated(propertyInfo.Property.SetMethod) [](start = 31, length = 120)

Does this mean read/write properties appear before any readonly properties? If so, won't that be confusing and lead to searching for properties?

Reply:
Ah, yes, I didn't realize. We should compare them the same.